Docker autoscaling

richardliaw opened the pull request:
Test

richardliaw committed:
Fix up docker usage

richardliaw committed:
Merge branch 'master' into docker_autoscaling

richardliaw committed:
Merge branch 'fix_autoscaling_tune' into docker_autoscaling

richardliaw committed:
lint

richardliaw committed:
lit

richardliaw committed:
revert

hartikainen commented on python/ray/autoscaler/commands.py:
```diff
@@ -422,6 +429,28 @@ def _exec(updater, cmd, screen, tmux, expect_error=False, port_forward=None):
+        target: target file on host
+        override_cluster_name: set the name of the cluster
+    """
+    config = yaml.load(open(config_file).read())
```
with open(config_file, 'r') as f:
    config = yaml.load(f)

hartikainen commented on python/ray/autoscaler/commands.py:
```diff
@@ -422,6 +429,28 @@ def _exec(updater, cmd, screen, tmux, expect_error=False, port_forward=None):
+        config["cluster_name"] = override_cluster_name
+    config = _bootstrap_config(config)
+    base_path = os.path.basename(target_script)
+    cname = config["docker"]["container_name"]
```
`container_name = ...`

richardliaw committed:
fix

richardliaw committed:
lint

hartikainen commented on python/ray/autoscaler/commands.py:
```diff
@@ -422,6 +429,28 @@ def _exec(updater, cmd, screen, tmux, expect_error=False, port_forward=None):
+    config = _bootstrap_config(config)
+    base_path = os.path.basename(target_script)
+    cname = config["docker"]["container_name"]
+    cmd = "docker cp {} {}:{}".format(target_script, cname, base_path)
```
You might want to wrap the target with quotes, so that paths with `~` work too:
```
    cmd = "docker cp {} {}:\"{}\"".format(target_script, cname, base_path)
```

richardliaw commented:
Try hiding a symlink between ray_results for Tune and a host-container mount, which allows for transparent rsync between driver and worker nodes.

vipulharsh committed:
Track newly created actor's parent actor (#5098)

* Track parent actor of actor

* Update src/ray/raylet/node_manager.cc

Co-Authored-By: Stephanie Wang <swang@cs.berkeley.edu>

* Update src/ray/raylet/node_manager.cc

Co-Authored-By: Stephanie Wang <swang@cs.berkeley.edu>

* fixing a comment

* Fixing typo in a comment

* capturing task_spec instead of actor_data

* adding const for some local variables

* changing an if else to else

* Linted version

* use updated method to create task from task_data

Change-Id: I9c1a65134dc23a2d175047e96b86ab9d9cf61971

* fixing linter issues

Change-Id: I1def06218130b399d2527b999258aecf9abb98dd

stephanie-wang committed:
Only get actor ID if actor task (#5180)

raulchen committed:
Fix Java MultithreadingTest (#5182)

richardliaw committed:
[tune] Fast Node Recovery (#5053)

richardliaw committed:
update mounting

richardliaw committed:
lint

richardliaw committed:
Update wheels to 0.8.0dev2 (#5186)

richardliaw committed:
Merge branch 'master' into docker_autoscaling

raulchen committed:
Fix: ServerCallFactory's destructor not marked as virtual (#5185)

richardliaw committed:
config files

pcmoritz committed:
Update arrow to include user defined status for plasma (#5156)

richardliaw committed:
nits

richardliaw committed:
nit

joneswong committed:
Keep parameter space noise consistent with action space noise (Fix 5173) (#5193)

*  make parameter space noise consistent with action space noise

*  modified according to lint check

*  indent

raulchen committed:
Fix a multithreading bug in grpc `ClientCall` (#5196)

raulchen committed:
Reconstruct failed actors without sending tasks. (#5161)

* fast reconstruct dead actors

* add test

* fix typos

* remove debug print

* small fix

* fix typos

* Update test_actor.py

simon-mo committed:
Improve release process 0.7.2 (#5187)

edoakes committed:
Remove dependencies from TaskExecutionSpecification (#5166)

kfstorm committed:
[Java worker] Refactor object store and worker context on top of core worker (#5079)

ericl committed:
[rllib] Fix rollout.py with tuple action space (#5201)

* fix it

* update doc too

* fix rollout

ericl committed:
[rllib] Remove nested import (#5204)

* remove nested import

* Update metrics.py

richardliaw committed:
Add heartbeat test + Fix monitor.py (#5191)

qxcv committed:
[rllib] Make RLLib handle zero-length observation arrays (#5208)

* [rllib] Make _summarize handle zero-len arrays

Fixes #5207

* [rllib] Make aligned_array() handle empty arrays

* [rllib] Conform with old yapf

joneswong committed:
Fixed inconsistent action placeholder (#5213)

joneswong committed:
Remove redundant scaler of l2 reg (#5172)

*  remove redundant scaler of l2 reg

*  lint formatted

* Update ddpg_policy.py

MQQ committed:
Improve memory check (#5216)

* Improve MemoryMonitor

- Add an env var to control the threshold.
- Use cgroup memory limit and usage for container environment.

* linting

* white space

* add comment

joneswong committed:
Enable seeding actors for reproducible experiments (#5197)

*  enable graph-level worker-specific seed

*  lint checked

*  revised according to eric's suggestions

*  revised accordingly and added a test case

*  formated

* Update test_reproducibility.py

* Update trainer.py

* Update rollout_worker.py

* Update run_rllib_tests.sh

* Update worker_set.py

micafan committed:
[GCS] introduce new gcs client and refactor actor table (#5058)

zhijunfu committed:
[direct call] add local plasma provider (#5184)

ericl committed:
[rllib] Move some inline defs to avoid deserialization errors (#5228)

* fix bug

* move metrics too

joneswong committed:
Removed the implicit sync barrier at the end of each training iteration (#5217)

*  removed sync barrier at the end of each training iteration

*  formatted

*  modify the comment according to current semantics

*  lint check

* Update trainer.py

ericl committed:
[rllib] MultiCategorical shouldn't return array for kl or entropy (#5215)

* wip

* fix

richardliaw committed:
[tune] Fixup exception messages (#5238)

ericl committed:
[rllib] Remove experimental eager support

richardliaw committed:
Improved KeyboardInterrupt Exception Handling (#5237)

richardliaw committed:
[autoscaler] Clean up error messages on setup failure (#5210)

vakker committed:
Ray namespace added for k8s (#4111)

* Ray namespace added for k8s

* Submit.yaml update with k8s namespace

* K8s deployment doc update with namespace

pschafhalter committed:
[sgd] Deprecate old distributed SGD implementation (#5160)

* Deprecate old distributed SGD implementation

* Update README

jovany-wang committed:
Fix the issue when passing multiple options in one string (#5241)

* Fix

* Fix linting

* Fix linting

* Address

* Fix test

stephanie-wang committed:
Leave `ray.wait` calls open until the task or actor exits (#5234)

* Regression test

* Split TaskDependencyManager::SubscribeDependencies into ray.get and ray.wait dependencies
- Some initial implementation

* unit test

* Improve unit tests for TaskDependencyManager

* Implement SubscribeWaitDependencies and UnsubscribeWaitDependencies, unit tests passing

* Add ray.wait python test for drivers that exit early

* Add WorkerID to Worker

* Update test to use two nodes

* Regression test for ray.wait passes

* Extend regression test to include ray.wait from an actor

* Fix ClientID and WorkerIDs

* lint

* lint

* Remove unnecessary ray_get argument

* fix build

stephanie-wang committed:
Add regression test for actor load balancing (#5224)

* Add regression test for actor load balancing

* Increase timeout

* Reduce number of nodes?

ericl committed:
[rllib] Fix trainer state restore (#5257)

ericl committed:
Fix two types of eviction hangs (#5225)

ericl committed:
[rllib] Add Keras LSTM example with ModelV2 (#5258)

ericl committed:
[rllib] Port DDPG to the build_tf_policy pattern (#5242)

jiangzihao2009 committed:
[gRPC] Migrate raylet client implementation to grpc (#5120)

ericl committed:
[rllib] ModelV2 support for pytorch (#5249)

stephanie-wang committed:
Increase timeout for `ray.wait` test (#5273)

* Increase test timeout for ray.wait

* make sure the actor is scheduled

micafan committed:
avoid copying ActorTableData when NodeMananger updates an actor to GCS (#5244)

antoine-galataud committed:
[rllib] Configure learner queue timeout (#5270)

* configure learner queue timeout

* lint

* use config

* fix method args order, add unit test

* fix wrong param name

raulchen committed:
Add CODEOWNERS file (#5259)

llan-ml committed:
[tune] configurable maximum length of trial identifier (#5287)

ls-daniel committed:
[autoscaler] Add a 'request_cores' function for manual autoscaling (#4754)

richardliaw committed:
[sgd] Example for Training (#5292)

richardliaw committed:
[tune] experiment_analysis split to Analysis (#5115)

richardliaw committed:
Retry and exception for hang on memory store full (#5143)

ericl committed:
[rllib] Document ModelV2 and clean up the models/ directory (#5277)

LorenzoCevolani committed:
Correctly setting the input to Train (#3853)

In the ResNetTrainActor class, the data are now exactly build using the Train flag for the cifar_input script.

richardliaw committed:
[autoscaler] Local YAML readability (#5290)

llan-ml committed:
[tune] support nested dictionaries for CSVLogger (#5295)

ericl committed:
Revert "[autoscaler] Clean up error messages on setup failure (#5210)" (#5299)

This reverts commit 7fc15dbf7fa0769eecbe1692ed3a97b5c8b918dd.

richardliaw committed:
[sgd] hotfix example failure (#5297)

* hotfix

* Update train_example.py

jovany-wang committed:
Fix releasing CPUs incorrectly when actor creation task blocked. (#5271)

* Fix

* Remove useless log

* Address

* Fix typo

* sleep

ericl committed:
[rllib] Better example rnn envs (#5300)

simon-mo committed:
Bump version to 0.7.3 (#5301)

michaelzhiluo committed:
[rllib] Importance Sampling and KL Loss for APPO (#5051)

simon-mo committed:
Bump version to 0.8.0.dev3 (#5308)

micafan committed:
Rename ClientTableData to GcsNodeInfo (#5251)

simon-mo committed:
Fix Redis Test (#5302)

zhijunfu committed:
Support direct actor call (#5183)

raulchen committed:
Submit task asynchronously from raylet client (#5313)

robertnishihara committed:
Fix bug in passing large arguments to tasks. (#5325)

stephanie-wang committed:
Lineage cache performance optimization to avoid duplicate GCS requests #5327

flying-mojo committed:
Fix Tuple spaces in rollout.py (#5332)

Make sure that the initial action is also properly flattened.

hartikainen committed:
[autoscaler] Change sys.exit(1) in update ssh_cmd (#5266)

simon-mo committed:
[Asyncio] Allow Async_API to init when loop is running (#5323)

layssi committed:
Added CARLA Community Example (#5333)

lukasfolle committed:
Fixed link in tune that was not working (#5331)

jichan3751 committed:
[sgd] Replaced class Resources in sgd with `use_gpu` (#5252)

ericl committed:
[rllib] Add rock paper scissors multi-agent example (#5336)

ConeyLiu committed:
Fix log monitor read error (#5221)

hartikainen committed:
[rllib] Feature/soft actor critic v2 (#5328)

* Add base for Soft Actor-Critic

* Pick changes from old SAC branch

* Update sac.py

* First implementation of sac model

* Remove unnecessary SAC imports

* Prune unnecessary noise and exploration code

* Implement SAC model and use that in SAC policy

* runs but doesn't learn

* clear state

* fix batch size

* Add missing alpha grads and vars

* -200 by 2k timesteps

* doc

* lazy squash

* one file

* ignore tfp

* revert done

richardliaw committed:
[tune] Distributed example + walkthrough (#5157)

simon-mo committed:
`ray stop` sends `SIGKILL` instead of `SIGTERM` (#5354)

zhijunfu committed:
[direct call]  In memory store (#5303)

Commit:
[tune] Fix small bug in experiment_analysis (#5365)

* [tune] Fix small bug in experiment_analysis

_experiment_state was not set as an attribute of ExperimentAnalysis

* Update python/ray/tune/analysis/experiment_analysis.py

Co-Authored-By: Richard Liaw <rliaw@berkeley.edu>

ericl committed:
Reduce Ray / RLlib startup messages (#5368)

jovany-wang committed:
Fix the issue of not initializing GLOG

richardliaw committed:
[tune] Pandas as soft dep

kfstorm committed:
Fix duplicated timeout logic in AbstractRayRuntime.get() (#5338)

ericl committed:
[rllib] Try moving RLlib to top level dir (#5324)

richardliaw committed:
[docs] rewrite (#5175)

ericl committed:
Send raylet error logs through the log monitor (#5351)

jiangzihao2009 committed:
Add common preprocessing for each request in node manager. (#5296)

richardliaw committed:
[docs] Hotfix for removing unneeded files (#5383)

mawright committed:
Custom action distributions (#5164)

* custom action dist wip

* Test case for custom action dist

* ActionDistribution.get_parameter_shape_for_action_space pattern

* Edit exception message to also suggest using a custom action distribution

* Clean up ModelCatalog.get_action_dist

* Pass model config to ActionDistribution constructors

* Update custom action distribution test case

* Name fix

* Autoformatter

* parameter shape static methods for torch distributions

* Fix docstring

* Generalize fake array for graph initialization

* Fix action dist constructors

* Correct parameter shape static methods for multicategorical and gaussian

* Make suggested changes to custom action dist's

* Correct instances of not passing model config to action dist

* Autoformatter

* fix tuple distribution constructor

* bugfix

raulchen committed:
Cap concurrent requests (#5341)

pcmoritz committed:
Ray projects schema and validation (#5329)

richardliaw committed:
[tune] Allow nested values in trial runner (#5346)

wsjeon committed:
MADDPG implementation in RLlib (#5348)

RehanSD committed:
[docs] remove table from walkthrough (#5389)

simon-mo committed:
Check upstream with `git remote` (#5377)

jovany-wang committed:
[ID Refactor] Refactor ActorID, TaskID and ObjectID (#5286)

* Refactor ActorID, TaskID on the Java side.

Left a TODO comment

WIP for ObjectID

ADD test

Fix

Add java part

Fix Java test

Fix

Refine test.

Enable test in CI

* Extra a helper function.

* Resolve TODOs

* Fix Python CI

* Fix Java lint

* Update .travis.yml

Co-Authored-By: Stephanie Wang <swang@cs.berkeley.edu>

* Address some comments.

Address some comments.

Add id_specification.rst

Reanme id_specification.rst to id_specification.md

typo

Address zhijun's comments.

Fix test

Address comments.

Fix lint

Address comments

* Fix test

* Address comments.

* Fix build error

* Update src/ray/design_docs/id_specification.md

Co-Authored-By: Stephanie Wang <swang@cs.berkeley.edu>

* Update src/ray/design_docs/id_specification.md

Co-Authored-By: Stephanie Wang <swang@cs.berkeley.edu>

* Update src/ray/design_docs/id_specification.md

Co-Authored-By: Stephanie Wang <swang@cs.berkeley.edu>

* Update src/ray/design_docs/id_specification.md

Co-Authored-By: Stephanie Wang <swang@cs.berkeley.edu>

* Update src/ray/design_docs/id_specification.md

Co-Authored-By: Stephanie Wang <swang@cs.berkeley.edu>

* Address comments

* Update src/ray/common/id.h

Co-Authored-By: Stephanie Wang <swang@cs.berkeley.edu>

* Update src/ray/common/id.h

Co-Authored-By: Stephanie Wang <swang@cs.berkeley.edu>

* Update src/ray/common/id.h

Co-Authored-By: Stephanie Wang <swang@cs.berkeley.edu>

* Update src/ray/design_docs/id_specification.md

Co-Authored-By: Hao Chen <chenh1024@gmail.com>

* Update src/ray/design_docs/id_specification.md

Co-Authored-By: Hao Chen <chenh1024@gmail.com>

* Address comments.

* Address comments.

* Address comments.

* Update C++ part to make sure task id is generated determantic

* WIP

* Fix core worker

* Fix Java part

* Fix comments.

* Add Python side

* Fix python

* Address comments

* Fix linting

* Fix

* Fix C++ linting

* Add JobId() method to TaskID

* Fix linting

* Update src/ray/common/id.h

Co-Authored-By: Hao Chen <chenh1024@gmail.com>

* Update java/api/src/main/java/org/ray/api/id/TaskId.java

Co-Authored-By: Hao Chen <chenh1024@gmail.com>

* Update java/api/src/main/java/org/ray/api/id/TaskId.java

Co-Authored-By: Hao Chen <chenh1024@gmail.com>

* Update java/api/src/main/java/org/ray/api/id/ActorId.java

Co-Authored-By: Hao Chen <chenh1024@gmail.com>

* Address comments

* Add DriverTaskId embeding job id

* Fix tests

* Add python dor_fake_driver_id

* Address comments and fix linting

* Fix CI

ericl committed:
[rllib] [docs] Add some architecture diagrams (#5390)

pcmoritz committed:
Fix compiler warnings and make warnings fatal (#5375)

ericl committed:
Fix setup (#5400)

simon-mo committed:
Silence some installation process for build from source (#5396)

richardliaw committed:
[tune,autoscaler] Test yaml, add better distributed docs (#5403)

ericl committed:
[rllib] Centralized critic / PPO example on TwoStepGame (#5392)

simon-mo committed:
[Project] Implementing Project CLI (#5397)

ericl committed:
Clean up top level Ray dir (#5404)

simon-mo committed:
Bump 0.8.0.dev2 -> 0.8.0.dev3 (#5409)

jredondopizarro committed:
[tune] Add hyperopt warm start feature (#5372)

ericl committed:
Allow `address` instead of `redis_address` (#5412)

* addr

* wip

* fix typo

* add to start

* switch to ray address for train

* say address

* disambiguate help

* comments 2

jichan3751 committed:
[sgd] Tune interface for Pytorch MultiNode SGD (#5350)

ericl committed:
[rllib] Fix output API when lz4 not installed (#5421)

ericl committed:
[rllib] Autoregressive action distributions (#5304)

Commit:
[tune] Allow relative local_dir at tune.run() (#4734)

ericl committed:
[hotfix] fix Travis action dist test (#5428)

zhijunfu committed:
Fix TestDirectActorTaskCrossNodesFailure test (#5406)

richardliaw committed:
[minor][tune] update pbt docs (#5420)

robertnishihara committed:
Don't stop Jupyter notebook in ray stop. (#5387)

adamochayon committed:
Rename function to make actor example correct (#5432)

TomVeniat committed:
[tune] Fix get_best_logdir behaviour (#5429)

* Fix get_best_logdir behaviour

* addtest

ericl committed:
[rllib] RLlib in 60 seconds documentation (#5430)

lisadunlap committed:
[tune] Implement BOHB (#5382)

nflu committed:
[tune] Reporter crash fix (#5426)

Co-authored-by: Richard Liaw <rliaw@berkeley.edu>

richardliaw committed:
[tune] Fix Travis Blocking (#5448)

richardliaw committed:
[tune] Quick Fix BOHB example (#5449)

richardliaw committed:
[tune] MLFlow Logger (#5438)

stephanie-wang committed:
Update the pull request template (#5460)

* Add a why header to the pull request template

* Update .github/PULL_REQUEST_TEMPLATE.md

Co-Authored-By: Edward Oakes <ed.nmi.oakes@gmail.com>

* Update .github/PULL_REQUEST_TEMPLATE.md

* Update .github/PULL_REQUEST_TEMPLATE.md

Co-Authored-By: Robert Nishihara <robertnishihara@gmail.com>

kfstorm committed:
[Java worker] Migrate task execution and submission on top of core worker (#5370)

mitchellstern committed:
Fix json.loads compatibility issue for Python 3.5 (#5466)

robertnishihara committed:
Add test for mutually recursive remote functions. (#5349)

micafan committed:
Make GCS Client thread-safe. (#5413)

raulchen committed:
increase timeout for test_actor_lifetime_load_balancing (#5463)

raulchen committed:
Fix test_logging_to_driver and test_not_logging_to_driver (#5462)

pcmoritz committed:
Revert raylet to worker GRPC communication back to asio (#5450)

edoakes committed:
Check for dead processes in blocked ray start (#5458)

edoakes committed:
Use subprocess.check_output in tests (#5465)

gregSchwartz18 committed:
[docs] Added Instructions for Slurm (#5467)

* Added Instructions for Slurm

Made in response to #826

jon-chuang committed:
[rllib] Add autoregressive KL (#5469)

simon-mo committed:
[Project] Add Basic Session CLI Commands (#5433)

holli committed:
Fixed few broken links in docs (#5477)

* hyperband link changed

* tuned_examples link fix

* doc lstm link fix

* kubernetes example link fix

pcmoritz committed:
[autoscaler] Fix ssh control path length issue (#5476)

edoakes committed:
Scale bazel HTTP timeout by 5x (#5482)

robertnishihara committed:
Add a script for benchmarking performance for Ray developers. (#5472)

micafan committed:
support for subscription to an actor (#5269)

raulchen committed:
Fix direct actor transport not treating some tasks as failed (#5464)

pcmoritz committed:
Fix impala stress test (#5491)

micafan committed:
modify ActorStateAccessor::AsyncGet callback (#5417)

zhijunfu committed:
Support multiple store providers in ObjectInterface (#5452)

pcmoritz committed:
[projects] Project examples and documentation (#5407)

ericl committed:
Ray, Tune, and RLlib support for memory, object_store_memory options (#5226)

richardliaw committed:
[tune] ASHA xgboost and lightgbm examples (#5500)

edoakes committed:
Batch fetch requests in core worker get (#5342)

gehring committed:
[rllib] Adds eager support with a generic `TFEagerPolicy` class (#5436)

kfstorm committed:
[Java] Fix getCurrentActorId in multi-threading scenario. (#5506)

raulchen committed:
[Java] Support calling functions returning void (#5494)

pcmoritz committed:
Bump version to 0.7.4 (#5474)

richardliaw committed:
[tune] Fix for keras threading (#5517)

kfstorm committed:
[Java] Destroy native core worker before killing ray processes (#5516)

raulchen committed:
[Java] Automatically clean up temp files. (#5507)

* Remove unused code

* Auto clean up temp files

* refine

* fix

* print warning

* fix

ericl committed:
Add docs for memory quota settings (#5441)

khardengi committed:
Fixed: missing brackets when appending proc info on OutOfMemory (#5530)

* Fixed: missing brackets when appending proc info on OutOfMemory

proc_stats.append was missing the set of brackets when adding a tuple to the list, which resulted in runtime error instead of correct Out of Memory message display.

* Update memory_monitor.py

ericl committed:
[rllib] Document on traj postprocess (#5532)

* document on traj postprocess

* shorten it

ericl committed:
[rllib] Enable object store memory limit by default (#5534)

pcmoritz committed:
[projects] Add named commands to sessions (#5525)

simon-mo committed:
Remove previous version of ray.serve (#5541)

ericl committed:
[rllib] Fix some eager execution regressions with 1.13 (#5537)

* fix bugs with 1.13

* allow disable

ericl committed:
fix test (#5544)

ericl committed:
Fix autoscaler format string for memory (#5542)

* add format string

* fix cast

idthanm committed:
[tune] TF2.0 TensorBoard support (#5547)

* Fix tensorboard log issue with tensorflow2.0

* tf2 support

pcmoritz committed:
Fix log files being opened as unicode files (#5545)

kfstorm committed:
[Java] `ObjectID::fromRandom` sets proper flags (#5548)

richardliaw committed:
[docs] Second push of changes (#5391)

pcmoritz committed:
Make RAY_CHECK for actor re-creation non-fatal (#5553)

pcmoritz committed:
Fix O(n^2) behavior in the log_monitor (#5569)

richardliaw committed:
[docs] Distributed Training Quickfix  (#5571)

ericl committed:
Fix rllib image in readme and doc typo (#5579)

* fix

* rlllig

ericl committed:
[rllib] Forgot to register param noise layer variables

pcmoritz committed:
Bump version string to 0.8.0.dev4 (#5523)

robertnishihara committed:
Update doc versions from 0.8.0.dev3 to 0.8.0.dev4. (#5585)

ericl committed:
Ref count objects created with ray.put (#5590)

* wip

* add weakref option

* tune util

* test ref

* doc

* fix tests

ericl committed:
Warn on resource deadlock; improve object store error messages (#5555)

* wip

* wip

* wip

* wip

* wip

* add impl

* second

* warn once

ericl committed:
[rllib] Add docs on policy.model (#5597)

richardliaw committed:
[docs] Add Pull Request Template Check (#5600)

* prtemplate

* fix

* Update PULL_REQUEST_TEMPLATE.md

* Update .github/PULL_REQUEST_TEMPLATE.md

Co-Authored-By: Edward Oakes <ed.nmi.oakes@gmail.com>

pcmoritz committed:
Fix impala stress test (#5596)

ericl committed:
[tune] Deprecate tune.function (#5601)

* remove tune function

* remove examples

* Update tune-usage.rst

edoakes committed:
Create session_latest symlink for Ray sessions (#5580)

suquark committed:
Fix DeprecationWarning (#5608)

richardliaw committed:
[tune] Add encoder for PBT (#5599)

* Add encoder

* Apply suggestions from code review

ericl committed:
Replace --redis-address with --address in test, docs, tune, rllib (#5602)

* wip

* add tests and tune

* add ci

* test fix

* lint

* fix tests

* wip

* sugar dep

suquark committed:
Fix the rllib-stack image display problem (#5612)

richardliaw committed:
[docs] Add a feedback form (#5610)

* Create a feedback form

* feedback

* fix

* fix

* fix

* wording

edoakes committed:
Simplify symlinking and don't print warnings (#5615)

micafan committed:
fix CallbackReply resize (#5589)

llan-ml committed:
Remove the unused argument `cusomt_loggers` in Experiment. (#5619)

edoakes committed:
Better error message for actor class inheritance (#5598)

edoakes committed:
Clean up Wait() and Get() in the core worker (#5556)

jichan3751 committed:
[training] Tensorflow interface for MultiNode SGD (#5440)

richardliaw committed:
[tune] Global checkpointing for tune at end (#5499)

edoakes committed:
Temporarily remove pytest-sugar dependency (#5627)

richardliaw committed:
[tune] Fix TB Memory Leak (#5629)

micafan committed:
Fix code style in unit test of GCS. (#5634)

llan-ml committed:
Lazily create summary writer for TF2 logger. (#5631)

richardliaw committed:
[tune] Update trainable docs and support hparams (#5558)

zhijunfu committed:
ignore object exists error for memory store provider (#5607)

edoakes committed:
Clean up Wait() in the core worker (#5628)

ericl committed:
[rllib] Revert [rllib] Port DDPG to the build_tf_policy pattern (#5626)

stephanie-wang committed:
Project fixes and cleanups (#5632)

mitchellstern committed:
Fix bug in ray.errors and update its default behavior (#5576)

devin-petersohn committed:
Remove Modin from Ray wheels. (#5647)

There are several reasons for this:

* We no longer support python2
* There should be only 1 way of installing Modin
  * Issue management on these wheels
* I have never heard of anyone using this feature
* It is rarely kept up to date
* Modin depends on specific versions of Ray because of past API changes

pcmoritz committed:
Update release documentation after 0.7.4 release (#5646)

* Update release documentation after 0.7.4 release

* update

* update

kfstorm committed:
`ray stop` kills processes more carefully (#5508)

richardliaw committed:
[tune] Fix numerical error (#5653)

kfstorm committed:
[Java] Support multiple workers in Java worker process (#5505)

ericl committed:
Consolidate and clean up documentation (#5645)

ericl committed:
[rllib] Fix TF2 import of EagerVariableStore (#5625)

pcmoritz committed:
Fix log monitor process error when attempting to read raylet PID. (#5655)

stephanie-wang committed:
[projects] Wrap ProjectDefinition in a class (#5654)

kfstorm committed:
Support metadata for passing by value task arguments (#5527)

ericl committed:
Add internal_api.pin_object_data() for pinning arbitrary object ids (#5637)

* add intenral pin method

* add pin

* update

robertnishihara committed:
Remove timeout in test_actor_lifetime_load_balancing. (#5659)

ericl committed:
[rllib] Improve accessing model state docs (#5656)

* [rllib] better model docs

* fix

* s

kfstorm committed:
[Java] Support direct actor call in Java worker (#5504)

richardliaw committed:
Update Cloudpickle (#5643)

simon-mo committed:
[Flaky tests] FIx test fork (#5671)

* Start testing test_fork

Maybe queue actor takes too long to initialize, that's why we are
seeing "Many python processes started" since most of the python
tasks are blocked on ray.get

* Add a comment

robertnishihara committed:
Fix PyPI version in readme. (#5662)

hershg committed:
[tune] Save and Restore for bayesopt  (#5623)

pcmoritz committed:
[Projects] Add "session execute" (#5681)

ericl committed:
Take into account queue length in autoscaling (#5684)

ericl committed:
[rllib] Eager execution for centralized critic example, fix simple optimizer for multiagent (#5683)

ericl committed:
Deflake cluster heartbeat test (#5552)

kiddyboots216 committed:
[rllib] Validate that entropy coeff is not an integer (#5687)

* Validate that entropy coeff is not an integer

Passing an integer value for entropy coeff such as 0 raises an error somewhere inside the TF policy graph, so this checks to make sure the entropy coeff is a float.

* Cast to float instead

Also move this check after the negative value check

edoakes committed:
Properly short circuit core worker Get() on exception (#5672)

edoakes committed:
Raise error if space in redis password (#5673)

kfstorm committed:
Replace NotImplementedException with UnsupportedOperationException (#5694)

richardliaw committed:
Merge branch 'master' into docker_autoscaling

richardliaw committed:
autoscaler

richardliaw committed:
nit
